leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/json/LwM2mNodeJsonEncoder.java

```diff
@@ -177,7 +177,21 @@ public void visit(LwM2mResource resource) {
 
         @Override
         public void visit(LwM2mResourceInstance instance) {
-            throw new UnsupportedOperationException("not yet implemented");
+            LOG.trace("Encoding resource instance {} into JSON", instance);
+            if (!requestPath.isResourceInstance()) {
+                throw new CodecException("Invalid request path %s for JSON resource instance encoding", requestPath);
+            }
+            baseName = new LwM2mPath(requestPath.getObjectId(),
```

Comment:
maybe we can go with `baseName = requestPath`;

and below using null when calling `resourceElement`

Reply:
we can't do "baseName = requestPath" as we need ResourcePath but not ResourceInstancePath

Reply:
Maybe the confusion is about `resourcePath` parameter of  `createJsonArrayEntry, the parameter name should rather be `name`. (should we rename it ?)

The path of the LwM2mNode is the concatenation of `baseName` + `name`.
So you can put the whole path in `baseName` or in `name` or split it in each attribute.

Now there is another tricky part : `TimestampedData`. Here you can have several value for the same `LwM2mNode`at  different timecode.
E.g : 
```json
       {
          "bn":"/1024/0/1/0","e":[
             {"v":22.9,"t":500},
             {"v":22.4,"t":510},
             {"v":24.1,"t":520}]
       }
```
So in this case the better way sounds to me to put the node path in `basename`.

Reply:
I think we should rename it to "name", sounds good to me.

Reply:
Add a `public void json_timestamped_resource_instances() throws CodecException`, my guess is that with the current code it will not work.

Reply:
Done